redwood-lazylayout-compose/src/commonTest/kotlin/app/cash/redwood/lazylayout/compose/LazyListTest.kt

```diff
@@ -84,188 +70,44 @@ class LazyListTest {
+        assertThat(lazyList.placeholder)
+          .isEqualTo(List(30) { TextValue(Modifier, "Placeholder") })
+        assertThat(lazyList.items)
+          .isEqualTo(List(expectedItemCount) { TextValue(Modifier, it.toString()) })
+      }
+    }
   }
 
-  @OptIn(RedwoodCodegenApi::class)
   @Test
   fun scrollPopulatedLazyColumn() = runTest {
-    val windowedLazyList = FakeWindowedLazyList()
-    val factories = TestSchemaWidgetFactories(
-      TestSchema = TestSchemaTestingWidgetFactory(),
-      RedwoodLayout = RedwoodLayoutTestingWidgetFactory(),
-      RedwoodLazyLayout = RedwoodLazyLayoutFakeWidgetFactory(windowedLazyList),
-    )
-    val snapshot = TestSchemaTester(factories) {
+    TestSchemaTester {
       setContent {
         LazyColumn(placeholder = { Text("Placeholder") }) {
           items(100) {
             Text(it.toString())
           }
         }
       }
-      awaitSnapshot()
-      windowedLazyList.updateViewport(50, 60)
-      awaitSnapshot()
-    }
 
-    assertThat(snapshot)
-      .containsExactly(
-        DefaultLazyListValue.copy(
-          itemsBefore = 35,
-          itemsAfter = 25,
-          placeholder = List(30) { TextValue(Modifier, "Placeholder") },
-          items = List(40) { TextValue(Modifier, (it + 35).toString()) },
-        ),
-      )
-  }
-}
+      with(awaitSnapshot()) {
+        val lazyList = single() as LazyListValue
+        lazyList.onViewportChanged(50, 60)
+      }
 
-@OptIn(RedwoodCodegenApi::class)
-private suspend fun <R> TestSchemaTester(
-  factories: TestSchemaWidgetFactories<WidgetValue> = TestSchemaWidgetFactories(
-    TestSchema = TestSchemaTestingWidgetFactory(),
-    RedwoodLayout = RedwoodLayoutTestingWidgetFactory(),
-    RedwoodLazyLayout = RedwoodLazyLayoutTestingWidgetFactory(),
-  ),
-  body: suspend TestRedwoodComposition<List<WidgetValue>>.() -> R,
-): R =
-  coroutineScope {
-    val container = MutableListChildren<WidgetValue>()
-    val tester = TestRedwoodComposition(
-      this,
-      factories,
-      container,
-    ) {
-      container.map { it.value }
-    }
-    try {
-      tester.body()
-    } finally {
-      tester.cancel()
+      with(awaitSnapshot()) {
+        val lazyList = single() as LazyListValue
+        assertThat(lazyList.itemsBefore).isEqualTo(35)
+        assertThat(lazyList.itemsAfter).isEqualTo(25)
+        assertThat(lazyList.placeholder)
+          .isEqualTo(List(30) { TextValue(Modifier, "Placeholder") })
+        assertThat(lazyList.items)
+          .isEqualTo(List(40) { TextValue(Modifier, (it + 35).toString()) })
+      }
     }
   }
-
-private class RedwoodLazyLayoutFakeWidgetFactory(
-  private val windowedLazyList: WindowedLazyList<WidgetValue>,
-) : RedwoodLazyLayoutWidgetFactory<WidgetValue> {
-  override fun LazyList() = windowedLazyList
-  override fun RefreshableLazyList() = error("unexpected call")
 }
-
-private class FakeWindowedLazyList : WindowedLazyList<WidgetValue>(NoOpListUpdateCallback) {
```

Comment:
Previously we used the compose-side LazyList to drive this on the widget-side.

With FakeProcessor we’ve got a test facet for the widget side and I don’t think this integration test is necessary.